tests/plots/test_declarative.py

```diff
@@ -437,6 +437,43 @@ def test_declarative_events():
     return pc.figure
 
 
+@pytest.mark.mpl_image_compare(remove_text=True, tolerance=0)
+@needs_cartopy
+def test_declarative_raster_events():
+    """Test that resetting traitlets properly propagates in RasterPlot()."""
+    data = xr.open_dataset(get_test_data('narr_example.nc', as_file_obj=False))
+
+    raster = RasterPlot()
+    raster.data = data
+    raster.field = 'Geopotential_height'
+    raster.level = 850 * units.hPa
+
+    img = ImagePlot()
+    img.data = data
+    img.field = 'Geopotential_height'
+    img.level = 800 * units.hPa
+    img.colormap = 'hot'
+    img.image_range = (-100, 200)
+
+    panel = MapPanel()
+    panel.area = 'us'
+    panel.projection = 'lcc'
+    panel.plots = [raster, img]
+
+    pc = PanelContainer()
+    pc.size = (8, 8.0)
+    pc.panels = [panel]
+    pc.draw()
+
+    # Update some properties to make sure it regenerates the figure
+    raster.level = 700 * units.hPa
+    raster.field = 'Temperature'
+    img.colormap = 'viridis'
```

Comment:
This would then be

```python
raster.colormap = 'virdis'
raster.colorbar = 'vertical'
```

Reply:
Yep! Talked this and your ImagePlot() comment over with Drew yesterday - apologies for not pushing the changes. Since colormap is highlighted here, I just wanted to note that during our meeting yesterday we found that the color of the images that are produced may vary in tests as opposed when not in the test environment. Best way to get around this is to ensure that the colormap is always specified (even if using 'viridis').